services/bls_aggregation/blsagg.go

```diff
@@ -0,0 +1,364 @@
+	// we read them in ProcessNewSignature from the main thread when we receive new signed tasks,
+	// and remove them from its respective goroutine when the task is completed or reached timeout
+	// we thus need a mutex to protect taskChans
+	taskChansMutex     sync.RWMutex
+	avsRegistryService avsregistry.AvsRegistryService
+	logger             logging.Logger
+}
+
+var _ BlsAggregationService = (*BlsAggregatorService)(nil)
+
+func NewBlsAggregatorService(avsRegistryService avsregistry.AvsRegistryService, logger logging.Logger) *BlsAggregatorService {
+	return &BlsAggregatorService{
+		ResponseC:          make(chan BlsAggregationServiceResponse),
+		taskChans:          make(map[types.TaskIndex]chan types.SignedTaskResponseDigest),
+		taskChansMutex:     sync.RWMutex{},
+		avsRegistryService: avsRegistryService,
+		logger:             logger,
+	}
+}
+
+func (a *BlsAggregatorService) GetResponseChannel() <-chan BlsAggregationServiceResponse {
+	return a.ResponseC
+}
+
+func (a *BlsAggregatorService) InitializeNewTask(
+	taskIndex types.TaskIndex,
+	taskCreatedBlock uint32,
+	quorumNumbers []types.QuorumNum,
+	quorumThresholdPercentages []types.QuorumThresholdPercentage,
+	timeToExpiry time.Duration,
+) error {
+	if _, taskExists := a.taskChans[taskIndex]; taskExists {
+		return TaskAlreadyInitializedErrorF(taskIndex)
+	}
+	taskC := make(chan types.SignedTaskResponseDigest)
+	a.taskChansMutex.Lock()
+	a.taskChans[taskIndex] = taskC
+	a.taskChansMutex.Unlock()
+	go a.singleTaskAggregatorGoroutineFunc(taskIndex, taskCreatedBlock, quorumNumbers, quorumThresholdPercentages, timeToExpiry, taskC)
+	return nil
+}
+
+func (a *BlsAggregatorService) ProcessNewSignature(
+	ctx context.Context,
+	taskIndex types.TaskIndex,
+	taskResponseDigest types.TaskResponseDigest,
+	blsSignature *bls.Signature,
+	operatorId bls.OperatorId,
+) error {
+	a.taskChansMutex.Lock()
+	taskC, taskInitialized := a.taskChans[taskIndex]
+	a.taskChansMutex.Unlock()
+	if !taskInitialized {
+		return TaskNotFoundErrorF(taskIndex)
+	}
+	signatureVerificationErrorC := make(chan error)
+	// send the task to the goroutine processing this task
+	// and return the error (if any) returned by the signature verification routine
+	select {
+	case taskC <- types.SignedTaskResponseDigest{
+		TaskResponseDigest:          taskResponseDigest,
+		BlsSignature:                blsSignature,
+		OperatorId:                  operatorId,
+		SignatureVerificationErrorC: signatureVerificationErrorC,
+	}:
+		// note that we need to wait synchronously here for this response because we want to
+		// send back an informative error message to the operator who sent his signature to the aggregator
+		return <-signatureVerificationErrorC
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+func (a *BlsAggregatorService) singleTaskAggregatorGoroutineFunc(
+	taskIndex types.TaskIndex,
+	taskCreatedBlock uint32,
+	quorumNumbers []types.QuorumNum,
```

Comment:
in the comment to this function, it should be mentioned that the `quorumNumbers` contain identification of the quorums whose threshold percentages are defined in `quorumThresholdPercentages`. Without this comment, the customer of the SDK might miss on this relationship.

Reply:
Added bunch of comments about this [0b6351d](https://github.com/Layr-Labs/eigensdk-go/pull/24/commits/0b6351d05b09cb7e98a3750d8822b36d27b7bc0b)